core/src/main/java/org/apache/iceberg/encryption/AesGcmInputFile.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.SeekableInputStream;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public class AesGcmInputFile implements InputFile {
+  private InputFile sourceFile;
+  private byte[] dataKey;
+  private long plaintextLength;
+
+  public AesGcmInputFile(InputFile sourceFile, byte[] dataKey) {
+    this.sourceFile = sourceFile;
+    this.dataKey = dataKey;
+    this.plaintextLength = -1;
+  }
+
+  @Override
+  public long getLength() {
+    Preconditions.checkArgument(plaintextLength >= 0, "Length is known after new stream is created");
```

Comment:
Should we set the length if field plaintextLength is -1? Looks like this method is valid only after `newStream()` is called.
```
if(plaintextLength == -1) {
this.newStream();
}
```

Reply:
When do we close the stream in this case? Or do we just make and close it in that case?

Reply:
Both suggestions sound good to me.